### DISCUSSION TITLE
Typescript client toJSON() and init() methods - make them optional or remove?

### DISCUSSION THREAD

**Discussion by terrencejames-lf** [RicoSuter/NSwag#4044] [Q&A] [answered]
I noticed some of the return/result classes in the Typescript client have this extra toJSON() and init() methods like this: 

```
export declare class Entry implements IEntry {
    /** The ID of the entry. */
    id?: number;
    /** The name of the entry. */
    name?: string | undefined;
    constructor(data?: IEntry);
    init(_data?: any): void;
    static fromJS(data: any): Entry;
    toJSON(data?: any): any;
}
```

However, when I try to instantiate objects of this class, I need to include these init and toJSON methods: 
![image](https://user-images.githubusercontent.com/101907777/172954995-0ac21745-8ed5-43c0-83a7-6b2ae2be039c.png)

This is cumbersome and not very user-friendly - is there a way to remove or make these two properties optional?

**Comment by lahma** [comment, marked as answer]
you should be able to initialize like this:

```typescript
const dummy = new Entry({
  id: 10,
  name: "Dummy"
});
```

Entry accepts the interface contract as constructor parameter.
